src/main/java/com/authservice/authservice/service/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.authservice.authservice.service;
+
+import com.authservice.authservice.dao.UserDetailsDao;
+import com.authservice.authservice.model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private static Logger logger = LoggerFactory.getLogger(UserDetailsServiceImpl.class);
+
+    private String url;
+
+    private UserDetailsDao userDetailsDao;
+
+    public UserDetailsServiceImpl(UserDetailsDao userDetailsDao) {
+        this.userDetailsDao = userDetailsDao;
+    }
+
+    /**
+     * This function loads the userDetails
+     *
+     * @param userName
+     * @return UserDetails
+     * @throws UsernameNotFoundException
+     */
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        logger.debug("Entering UserDetailsServiceImpl.loadUserByUsername with parameter userName {}.", userName);
+        User user = userDetailsDao.getUserByUsername(userName);
+        if (user == null) {
+            throw new UsernameNotFoundException("User not found with username: " + userName);
```

Comment:
should ideally be `NotFoundException` instead of creating a new class `UsernameNotFoundException`

Reply:
It is a Spring Security exception.